pages/settings/SettingsLayout.qml

```diff
@@ -177,7 +278,43 @@ Rectangle {
         }
     }
 
+    ListModel {
+        id: fiatPriceProvidersModel
+    }
+
+    ListModel {
+        id: fiatPriceCurrencyModel
+        ListElement {
+            data: "xmrusd"
+            column1: "USD"
+        }
+        ListElement {
+            data: "xmreur"
+            column1: "EUR"
+        }
+    }
+
     Component.onCompleted: {
+        // Dynamically fill fiatPrice dropdown based on `appWindow.fiatPriceAPIs`
+        var apis = appWindow.fiatPriceAPIs;
+        fiatPriceProvidersModel.clear();
+
+        var i = 0;
+        for (var api in apis){
+            if (!apis.hasOwnProperty(api))
+               continue;
+
+            fiatPriceProvidersModel.append({"column1": Utils.capitalize(api), "data": api});
+
+            if(api === persistentSettings.fiatPriceProvider)
+                fiatPriceProviderDropDown.currentIndex = i;
+            i += 1;
+        }
+
+        fiatPriceProviderDropDown.update();
```

Comment:
Is update twice intentional?

Reply:
`fiatPriceProviderDropDown` && `fiatPriceCurrencyDropdown `